js/gantt-helper.js

```diff
@@ -0,0 +1,893 @@
+      { name: "time", type: "duration", map_to: "auto" },
+      { name: "type", type: "radio", map_to: "type", options: project_subtypes, default_value: gantt.config.types.project }
+   ];
+
+   var new_item_section = [
+      { name: "description", height: 70, map_to: "text", type: "textarea", focus: true },
+      { name: "time", type: "duration", map_to: "auto" },
+      { name: "type", type: "radio", map_to: "type", options: task_subtypes, default_value: gantt.config.types.task }
+   ];
+
+   var filterValue = "";
+   var delay;
+
+   // >>>>> Configs
+   gantt.config.grid_width = 600;
+   gantt.config.grid_resize = true;
+   gantt.config.date_format = "%Y-%m-%d %H:%i";
+
+   gantt.config.order_branch = "marker";
+   gantt.config.order_branch_free = true;
+   gantt.config.show_progress = true;
+   gantt.config.sort = true;
+
+   gantt.config.lightbox.project_sections = [
+      { name: "description", height: 70, map_to: "text", type: "textarea", focus: true },
+   ];
+
+   gantt.config.lightbox.sections = [
+      { name: "description", height: 70, map_to: "text", type: "textarea", focus: true },
+      { name: "time", type: "duration", map_to: "auto" }
+   ];
+
+   gantt.config.lightbox.milestone_sections = [
+      { name: "description", height: 70, map_to: "text", type: "textarea", focus: true },
+      { name: "time", type: "duration", single_date: true, map_to: "auto" }
+   ];
+
+   // disable task specific controls on projects
+   gantt.templates.task_class = function(start, end, task) {
+      var css = [];
+      if (task.type == "project") {
+         css.push("no_progress_drag");
+         css.push("no_link_drag");
+      }
+      return css.join(" ");
+   };
+
+   // set text labels for milestones
+   gantt.templates.rightside_text = function(start, end, task) {
+      if (task.type == gantt.config.types.milestone) {
+         return task.text;
+      }
+      return "";
+   };
+
+   gantt.templates.progress_text = function(start, end, task) {
+      return "<span style='text-align:left; color: #fff;'>" + Math.round(task.progress * 100) + "% </span>";
+   };
+
+   // enable tooltips and fullscreen mode
+   gantt.plugins({ tooltip: true, fullscreen: true });
+
+   gantt.templates.tooltip_text = function(start, end, task) {
+      var text = "<b><span class=\"capitalize\">" +
+         task.type + ":</span></b> " + task.text + "<br/><b>Start date:</b> " +
+         gantt.templates.tooltip_date_format(start) +
+         "<br/><b>End date:</b> " + gantt.templates.tooltip_date_format(end) +
+         "<br/><b>Progress:</b> " + parseInt(task.progress * 100) + "%";
+      if (task.content && task.content.length > 0) {
+         text += "<br/><b>Description:</b><div style=\"padding-left:25px\">" + task.content + "</div>";
+      }
+      if (task.comment && task.comment.length > 0) {
+         text += "<br/><b>Comment:</b><div style=\"padding-left:25px\"" + task.comment + "</div>";
+      }
+      return text;
```

Comment:
Ok, for ajax messages if I put something like
`gantt.alert(__('Could not update Task[%s]: ', id) + json.error);`
how can I update the language files (mo, po) to have this message working ?

Reply:
In current PR, you do not have to update po/mo files. It will be done separately.

For your GLPI instance, you can manually create a file using POEdit, and store corresponding mo file in `files/core/xx_YY.mo`. You may have to clear GLPI cache and restart your PHP server.